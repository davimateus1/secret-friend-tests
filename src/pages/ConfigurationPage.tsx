import { Flex, Text } from '@chakra-ui/react';
import { Card, Footer, ParticipantsList, UserForm } from '../components';

export const ConfigurationPage = () => {
  return (
    <Card>
      <Flex direction="column" w="100%">
        <Text
          fontWeight="bold"
          fontSize="4xl"
          textAlign="center"
          mb="3rem"
          color="#4B69FD"
        >
          Vamos come√ßar!
        </Text>
        <UserForm />
        <ParticipantsList />
        <Footer />
      </Flex>
    </Card>
  );
};
