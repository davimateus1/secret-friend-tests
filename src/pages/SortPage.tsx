import { Card } from '../components';
import Airplane from '../assets/airplane.png';

import {
  ChangeEvent,
  FormEvent,
  useCallback,
  useEffect,
  useState,
} from 'react';

import { useNavigate } from 'react-router-dom';
import { useParticipantsList, useSortResult } from '../state';
import { Button, Flex, Image, Select, Text } from '@chakra-ui/react';

export const SortPage = () => {
  const result = useSortResult();
  const navigate = useNavigate();
  const participants = useParticipantsList();

  const [secretFriend, setSecretFriend] = useState('');
  const [actualParticipant, setActualParticipant] = useState('');

  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (result.has(actualParticipant)) {
      const secretFriend = result.get(actualParticipant)!;
      setSecretFriend(secretFriend);
    }
  };

  const storeResult = useCallback(() => {
    const resultObj = Object.fromEntries(result);
    const resultJSON = JSON.stringify(resultObj);

    localStorage.setItem('result', resultJSON);
  }, [result]);

  const handleChangeParticipant = (e: ChangeEvent<HTMLSelectElement>) => {
    setActualParticipant(e.target.value);
  };

  useEffect(() => {
    if (result.size) return storeResult();
    navigate('/');
  }, [result, participants, storeResult, navigate]);

  return (
    <Card>
      <form onSubmit={handleSubmit}>
        <Flex fontSize="xl" direction="column" align="center">
          <Text
            fontWeight="bold"
            fontSize="4xl"
            textAlign="center"
            mb="3rem"
            color="brand.900"
          >
            Quem vai tirar o papelzinho?
          </Text>
          <Select
            required
            id="sortedParticipant"
            name="sortedParticipant"
            value={actualParticipant}
            onChange={handleChangeParticipant}
            w="100%"
            h="6rem"
            fontSize="2xl"
            borderRadius="4rem"
            boxSizing="border-box"
            border="2px solid black"
            data-testid="select-participant"
            boxShadow="0px 2px 0px 1px black"
            _focus={{ outline: 'none' }}
          >
            <option value="" disabled>
              Selecione o participante
            </option>
            {participants.map((participant) => (
              <option key={participant} value={participant}>
                {participant}
              </option>
            ))}
          </Select>
          <Text mt="1rem" mb="2rem" textAlign="center">
            Clique em sortear para ver quem é o seu amigo secreto!
          </Text>
          <Button
            h="6rem"
            w="70%"
            m="1rem 0"
            type="submit"
            color="white"
            bg="brand.800"
            fontSize="2xl"
            cursor="pointer"
            borderRadius="4rem"
            _hover={{ bg: 'brand.900' }}
            boxShadow="2px 2px 0px 1px black"
            _disabled={{ opacity: 0.6, cursor: 'not-allowed' }}
            id="sort-button"
          >
            Sortear
          </Button>
          <Text
            m="2rem 0"
            role="alert"
            fontSize="2xl"
            fontWeight="500"
            color="brand.800"
            textAlign="center"
            id="sort-result"
          >
            {secretFriend ? `Seu amigo secreto é: ${secretFriend}` : ''}
          </Text>
          <Flex color="brand.800" fontSize="xl" justify="center">
            <Image src={Airplane} alt="Airplane" w="11rem" />
          </Flex>
        </Flex>
      </form>
    </Card>
  );
};
