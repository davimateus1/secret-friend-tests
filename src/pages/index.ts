export * from './SortPage';
export * from './ConfigurationPage';
