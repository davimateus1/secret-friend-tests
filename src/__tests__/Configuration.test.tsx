import { RecoilRoot } from 'recoil';
import { ConfigurationPage } from '../pages';
import { render } from '@testing-library/react';

const mockNavigation = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigation,
  };
});

describe('Configuration Page', () => {
  it('Should render the configuration page', () => {
    const { container } = render(<ConfigurationPage />, {
      wrapper: RecoilRoot,
    });
    expect(container).toMatchSnapshot();
  });
});
