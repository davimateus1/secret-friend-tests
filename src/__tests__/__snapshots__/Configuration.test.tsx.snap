// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration Page Should render the configuration page 1`] = `
<div>
  <div
    class="css-1ne9sec"
  >
    <div
      class="css-1ofqig9"
    >
      <p
        class="chakra-text css-vtxltj"
      >
        Vamos come√ßar!
      </p>
      <form>
        <div
          class="css-2ygrde"
        >
          <input
            class="chakra-input css-1jda5v6"
            id="add-participant-input"
            placeholder="Insira os nomes dos participantes"
            value=""
          />
          <button
            class="chakra-button css-13fehna"
            disabled=""
            id="add-participant-button"
            type="submit"
          >
            Adicionar
          </button>
        </div>
      </form>
      <ul
        class="css-hw3umf"
        id="participants-list"
        role="list"
      />
      <div
        class="css-neqxv"
      >
        <button
          class="chakra-button css-1uyux1n"
          disabled=""
          id="start-button"
          type="button"
        >
          Iniciar brincadeira!
        </button>
        <img
          alt="Bags"
          class="chakra-image css-0"
          src="bags.png"
        />
      </div>
    </div>
  </div>
</div>
`;
